tests/spread/core22/squid-proxy/task.yaml

```diff
@@ -0,0 +1,53 @@
+summary: Verify that squid proxy can be passed as an http proxy to snapcraft.
+
+environment:
+  SNAP_DIR: snaps/squid-proxy-test
+  SQUID_PROXY_CONF: /etc/squid/squid.conf
+  SQUID_PROXY_LOG: /var/log/squid/access.log
+
+prepare: |
+  sudo apt-get install --yes squid
+
+  # Configure squid
+  sed -i 's|^http_access deny CONNECT !SSL_ports|#http_access deny CONNECT !SSL_ports|' ${SQUID_PROXY_CONF}
+  sed -i 's|^# http_access allow localnet|http_access allow localnet|' ${SQUID_PROXY_CONF}
+  sed -i 's|^# via on|via off|' ${SQUID_PROXY_CONF}
+  sed -i 's|^# forwarded_for on|forwarded_for delete|' ${SQUID_PROXY_CONF}
+
+  sudo systemctl restart squid.service
+
+restore: |
+  cd "$SNAP_DIR"
+  snapcraft clean
+  rm -f ./*.snap
+
+  sudo apt-get remove --purge --yes squid
+
+execute: |
+  cd "$SNAP_DIR"
+
+  SQUID_PROXY_URL="http://$(hostname -I | awk '{print $1}'):3128"
```

Comment:
Wouldn't localhost be a more obvious candidate to use here?

Reply:
At the end of the day what we really want is the host IP. Using ~ `SQUID_PROXY_URL=localhost`, this gets propagated as is to the LXD container and we end up with `localhost` referring to the container, not the host.
I tried using the DNS entry `_gateway.lxd` that resolves to the host through LXD built-in DNS. It works locally but seems to only resolve to the IPv6 address once in CI. Configuring squid proxy for IPv6 is a different business, one not worth tackling for a 'simple' test.

If you have any other suggestion to get the host IPv4 address, I'm all ears.

Reply:
I was just curious, I think it's fine this way.